Arrowgene.Ddon.GameServer/Handler/PawnCreatePawnHandler.cs

```diff
@@ -0,0 +1,350 @@
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Network;
+using Arrowgene.Logging;
+using System.Collections.Generic;
+using System;
+using System.Linq;
+using Arrowgene.Ddon.GameServer.Characters;
+
+namespace Arrowgene.Ddon.GameServer.Handler
+{
+    public class PawnCreatePawnHandler : GameRequestPacketHandler<C2SPawnCreatePawnReq, S2CPawnCreatePawnRes>
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(PawnCreatePawnHandler));
+
+        public PawnCreatePawnHandler(DdonGameServer server) : base(server)
+        {
+        }
+
+        public override S2CPawnCreatePawnRes Handle(GameClient client, C2SPawnCreatePawnReq request)
+        {
+            Pawn pawn = new Pawn(client.Character.CharacterId)
+            {
+                Name = request.PawnInfo.Name,
+                EditInfo = request.PawnInfo.EditInfo,
+                HideEquipHead = request.PawnInfo.HideEquipHead,
+                HideEquipLantern = request.PawnInfo.HideEquipLantern,
+                Job = request.PawnInfo.JobId,
+                HmType = 1,
+                PawnType = 1,
+                ExtendedParams = new CDataOrbGainExtendParam(),
+                Server = client.Character.Server
+            };
+            PopulateNewPawnData(pawn);
+            Server.CharacterManager.UpdateCharacterExtendedParams(pawn, true);
+
+            if (request.SlotNo > 1)
+            {
+                // We need to consume items for the cost
+                var uidItem = new Item() { ItemId = 10133 };
+                Server.ItemManager.ConsumeItemByUIdFromItemBag(Server, client.Character, uidItem.UId, 10);
```

Comment:
Is it always item bag? What if it's in the storage box?

Reply:
Good quest, I am not sure. You don't actually select the items. The way the menu works, it just says you have enough or it tells you you don't have enough. If you have enough and press ok, it transitions to the pawn create menu.

Reply:
Does it allow you to go through if you have them in the storage box, expansion, or arisen room storage chest?

Reply:
It looks like any storage counts. I just moved it from my bag to the box and the menu option still pops up.

![image](https://github.com/user-attachments/assets/57c61bdd-69d3-4d1f-9423-caad71e7c7cc)

Reply:
What if there's less than 10 in each storage but it all adds up to 10? Either way, ConsumeItemByUIdFromMultipleStorages should probably be used in it's place

Reply:
I split my stack up into 2 stacks. One in my item back and one in box normal storage and the option still shows up. So seems the client just knows we have enough.